modules/cudacodec/src/video_reader.cpp

```diff
@@ -345,14 +351,14 @@ namespace
             else
                 break;
         case VideoReaderProps::PROP_NUMBER_OF_RAW_PACKAGES_SINCE_LAST_GRAB:
-            propertyVal = rawPackets.size();
+            propertyVal = static_cast<double>(rawPackets.size());
             return true;
         case VideoReaderProps::PROP_RAW_MODE:
             propertyVal = videoSource_->RawModeEnabled();
             return true;
         case VideoReaderProps::PROP_LRF_HAS_KEY_FRAME: {
-            const int iPacket = propertyVal - rawPacketsBaseIdx;
-            if (videoSource_->RawModeEnabled() && iPacket >= 0 && iPacket < rawPackets.size()) {
+            const int iPacket = static_cast<int>(propertyVal) - rawPacketsBaseIdx;
```

Comment:
`propertyVal` is function output and could contain any value here. Looks like it's a bug.

Reply:
Its output and input, not sure what I was thinking here will alter this so that it makes more sense.

Reply:
Please document "input" logic then.

Reply:
@asmorkalov I have added a new method `rawPackageHasKeyFrame()` to avoid the input output argument confusion in the `get()` method but now I am wondering if it is the right time to make some more alterations.  Specifically those below which I "wrongly" introduced to mirror the `VideoCapture` class:

- remove `grab()/retrieve()` for the decoded frame.  This combination is completely unecessary `grab()` doesn't save any processing, it only remove the need to pass a `GpuMat` when you don't want to retrieve the decoded frame.
- `propertyVal` of `double` in `get(const VideoReaderProps propertyId, CV_OUT double& propertyVal)` results in a lot of needless casting when the value will always be an integer.

What do you think?

Reply:
@asmorkalov I have pushed changes to demonstrate the removal of `grab()` and making `propertyVal` a `size_t` what do you think?